ako-gateway-api/lib/dynamic_client.go

```diff
@@ -104,57 +103,72 @@ func GetDynamicInformers() *DynamicInformers {
 	return dynamicInformerInstance
 }
 
-func ParseL7CRD(key, namespace, name string, vsNode nodes.AviVsEvhSniModel, isFilterAppProfSet bool) error {
-
+func IsL7CRDValid(key, namespace, name string) (bool, *unstructured.Unstructured, error) {
 	clientSet := GetDynamicClientSet()
 	if clientSet == nil {
-		return fmt.Errorf("key: %s, msg:error in fetching L7Rule CRD object", key)
+		return false, nil, fmt.Errorf("error in fetching L7Rule CRD object. clientset is nil")
 	}
 	obj, err := clientSet.Resource(L7CRDGVR).Namespace(namespace).Get(context.TODO(), name, metav1.GetOptions{})
 	if err != nil {
 		if k8serrors.IsNotFound(err) {
-			return fmt.Errorf("key: %s, msg: error: L7Rule CRD %s/%s not found", key, namespace, name)
+			utils.AviLog.Errorf("key:%s/%s, msg:L7Rule CRD not found: %+v", namespace, name, err)
```

Comment:
Should we be using the key in the parameter instead of creating newly?

Reply:
will take care of it in later PRs